stream/src/utils/codec.rs

```diff
@@ -36,11 +37,16 @@ pub async fn recv_frame<T: Stream>(
     max_message_size: usize,
 ) -> Result<Bytes, Error> {
     // Read the first 4 bytes to get the length of the message
-    let mut buf = [0u8; 4];
-    stream.recv(&mut buf).await.map_err(Error::RecvFailed)?;
+    let mut read = stream
+        .recv(size_of::<u32>())
```

Comment:
`u32` should have a `u32::FIXED_SIZE` I think?

We've moved away from the `size_of` usage generally.